### PR TITLE
GROOVY-7952: Property expressions for extension methods starting with…

### PR DESCRIPTION
**Pull request by shils**
… 'is' fail STC

* Add support for 'is' getter method variants of GROOVY-5580

### CODE REVIEW

**Comment by blackdrag** [comment]
should we limit the lookup of is-property-accessor-methods to boolean returning ones?

**Comment by shils** [comment]
Agreed, the lookups should match groovy's generation of synthetic methods.

**Comment by paulk-asert** [comment]
+1 with the check for boolean

**Comment by shils** [comment]
Amended to reflect @blackdrag 's comment

File: src/main/org/codehaus/groovy/classgen/asm/sc/StaticTypesCallSiteWriter.java [hunk 1]
```diff
@@ -158,19 +158,24 @@ public void makeGetPropertySite(Expression receiver, final String methodName, fi
...
+                if (getterMethod == null) getterMethod = anInterface.getGetterMethod(altGetterName);
+                if (getterMethod != null) break;
             }
             // GROOVY-5585
-            if (getterMethod==null) {
+            if (getterMethod == null) {
                 getterMethod = OBJECT_TYPE.getGetterMethod(getterName);
             }
+            if (getterMethod == null) {
+                getterMethod = OBJECT_TYPE.getGetterMethod(altGetterName);
```
Comment (paulk-asert) [inline src/main/org/codehaus/groovy/classgen/asm/sc/StaticTypesCallSiteWriter.java]: There are no isXxx methods in Object? Is this needed?

Comment (shils) [inline src/main/org/codehaus/groovy/classgen/asm/sc/StaticTypesCallSiteWriter.java]: I suppose not, removed.

**Pull request merged**